next/CMakeLists.txt

```diff
@@ -0,0 +1,940 @@
+    ${MBGL_ROOT}/src/parsedate/parsedate.hpp
+)
+
+if(EXISTS ${MBGL_ROOT}/.git/HEAD)
+    execute_process(
+        COMMAND
+            git
+            rev-parse
+            --short=8
+            HEAD
+        WORKING_DIRECTORY ${MBGL_ROOT}
+        OUTPUT_VARIABLE MBGL_VERSION_REV
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
+else()
+    set(MBGL_VERSION_REV 00000000)
+endif()
+
+set_source_files_properties(
+    ${MBGL_ROOT}/src/mbgl/util/version.cpp
+    PROPERTIES
+    COMPILE_DEFINITIONS
+    MBGL_VERSION_REV="${MBGL_VERSION_REV}"
+)
+
+target_include_directories(
+    mbgl-core
+    PRIVATE ${MBGL_ROOT}/src
+)
+
+target_include_directories(
+    mbgl-core
+    PUBLIC ${MBGL_ROOT}/include
+)
+
+add_subdirectory(${PROJECT_SOURCE_DIR}/vendor/mapbox-base/mapbox)
+add_subdirectory(${PROJECT_SOURCE_DIR}/vendor/mapbox-base/extras)
+
+include(${PROJECT_SOURCE_DIR}/vendor/boost.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/earcut.hpp.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/eternal.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/expected.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/geojson-vt-cpp.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/polylabel.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/protozero.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/shelf-pack-cpp.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/unique_resource.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/vector-tile.cmake)
+include(${PROJECT_SOURCE_DIR}/vendor/wagyu.cmake)
+
+target_link_libraries(
+    mbgl-core
+    PRIVATE
+        Mapbox::Base::Extras::kdbush.hpp
+        Mapbox::Base::supercluster.hpp
+        mbgl-vendor-boost
+        mbgl-vendor-earcut.hpp
+        mbgl-vendor-eternal
+        mbgl-vendor-geojson-vt-cpp
+        mbgl-vendor-polylabel
+        mbgl-vendor-protozero
+        mbgl-vendor-shelf-pack-cpp
+        mbgl-vendor-unique_resource
+        mbgl-vendor-vector-tile
+        mbgl-vendor-wagyu
+    PUBLIC
+        Mapbox::Base::Extras::rapidjson
+        Mapbox::Base::geojson.hpp
+        Mapbox::Base::geometry.hpp
+        Mapbox::Base::optional
+        Mapbox::Base::typewrapper
+        Mapbox::Base::variant
+        Mapbox::Base::weak
+        mbgl-vendor-expected
+)
+
+set_property(TARGET mbgl-core PROPERTY FOLDER Core)
+
+add_library(
+    Mapbox::Map ALIAS mbgl-core
+)
+
+if(MBGL_WITH_CORE_ONLY)
+    return()
+elseif(MBGL_WITH_QT)
+    include(${PROJECT_SOURCE_DIR}/platform/qt/qt.cmake)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Android)
+    include(${PROJECT_SOURCE_DIR}/platform/android/android.cmake)
+elseif(CMAKE_SYSTEM_NAME STREQUAL iOS)
+    include(${PROJECT_SOURCE_DIR}/platform/ios/ios.cmake)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Linux)
+    include(${PROJECT_SOURCE_DIR}/platform/linux/linux.cmake)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    include(${PROJECT_SOURCE_DIR}/platform/macos/macos.cmake)
+else()
+    message(FATAL_ERROR "Unsupported target platform: " ${CMAKE_SYSTEM_NAME})
+endif()
+
+add_subdirectory(${PROJECT_SOURCE_DIR}/test)
```

Comment:
EXCLUDE_FROM_ALL ?

Reply:
it can't be, in the context of GL Native we want to build the tests always.

Reply:
To clarify `EXCLUDE_FROM_ALL` is used for auxiliary targets you don't need to know they exist, usually dependencies.

Reply:
> it can't be, in the context of GL Native we want to build the tests always.

Why is that? If that's because of CI we can make the test targets explicit.

Reply:
Also remember that in contexts where we're adding GL Native as a submodule we don't want to have GL native tests being built by default.